core/src/main/java/org/sonarsource/sonarlint/core/container/global/NodeJsHelper.java

```diff
@@ -0,0 +1,171 @@
+  private static final Pattern NODEJS_VERSION_PATTERN = Pattern.compile("v?(\\d+\\.\\d+\\.\\d+(.\\\\d+)?(-.*)?)");
+  private final System2 system2;
+  private final AbstractGlobalConfiguration config;
+  private final Path pathHelperLocationOnMac;
+  private final CommandExecutor commandExecutor;
+
+  private Path nodeJsPath;
+  private Version nodeJsVersion;
+
+  public NodeJsHelper(AbstractGlobalConfiguration config, System2 system2) {
+    this(config, system2, Paths.get("/usr/libexec/path_helper"), CommandExecutor.create());
+  }
+
+  // For testing
+  NodeJsHelper(AbstractGlobalConfiguration config, System2 system2, Path pathHelperLocationOnMac, CommandExecutor commandExecutor) {
+    this.config = config;
+    this.system2 = system2;
+    this.pathHelperLocationOnMac = pathHelperLocationOnMac;
+    this.commandExecutor = commandExecutor;
+  }
+
+  @Override
+  public void start() {
+    nodeJsPath = locateNode(config);
+    if (nodeJsPath != null) {
+      LOG.debug("Checking node version...");
+      Command command = Command.create(nodeJsPath.toString()).addArgument("-v");
+      String nodeVersionStr = runSimpleCommand(command);
+      if (nodeVersionStr != null) {
+        Matcher matcher = NODEJS_VERSION_PATTERN.matcher(nodeVersionStr);
+        if (matcher.matches()) {
+          String version = matcher.group(1);
+          nodeJsVersion = Version.create(version);
+          LOG.debug("Detected node version: {}", nodeJsVersion);
+        } else {
+          LOG.debug("Unable to parse node version: {}", nodeVersionStr);
+        }
+      }
+      if (nodeJsVersion == null) {
+        LOG.warn("Unable to query node version");
+      }
+    }
+  }
+
+  @CheckForNull
+  public Path getNodeJsPath() {
+    return nodeJsPath;
+  }
+
+  @CheckForNull
+  public Version getNodeJsVersion() {
+    return nodeJsVersion;
+  }
+
+  @CheckForNull
+  private Path locateNode(AbstractGlobalConfiguration config) {
+    Path configuredNodejsPath = config.getNodejsPath();
+    if (configuredNodejsPath != null) {
+      LOG.debug("Node.js path provided by configuration: {}", configuredNodejsPath);
+      return configuredNodejsPath;
+    }
+    LOG.debug("Looking for node in the PATH");
+
+    String result;
+    if (system2.isOsWindows()) {
+      result = runSimpleCommand(Command.create("where").addArgument("node"));
+    } else {
+      Command which = Command.create("which").addArgument("node");
+      if (system2.isOsMac() && Files.exists(pathHelperLocationOnMac)) {
+        Command command = Command.create(pathHelperLocationOnMac.toString()).addArgument("-s");
+        String pathHelperOutput = runSimpleCommand(command);
+        if (pathHelperOutput != null) {
+          Pattern regex = Pattern.compile(".*PATH=\"(.*)\"; export PATH;.*");
+          Matcher matchResult = regex.matcher(pathHelperOutput);
+          if (matchResult.matches()) {
+            which.setEnvironmentVariable("PATH", matchResult.group(1));
+          }
+        }
+      }
```

Comment:
For clarity (+ cognitive complexity) I'd extract this to a method, e.g `fixPathForMacOs`